frontend/include/chpl/uast/compiler-globals-list.h

```diff
@@ -19,6 +19,7 @@
 
 COMPILER_GLOBAL(bool, "boundsChecking", boundsChecking)
 COMPILER_GLOBAL(bool, "castChecking", castChecking)
+COMPILER_GLOBAL(bool, "constArgChecking", constArgChecking)
```

Comment:
I'm not sure why this is needed. Doesn't this make a symbol called `constArgChecking` available for use in user/module code? I don't seeany uses/tests of this in this PR. But maybe you are just planning ahead?

Reply:
I was doing a search for how similar flags were treated and guessed that this was something dyno needed to understand it, so branch-predicted instead of waiting for the failure.  I'm happy to remove it, though - I don't have a strong opinion for whether this is something that should be checkable in user/module code

Reply:
IMO we should remove it until it is needed, or we have an expectation of that.

FWIW it is also probably going to be needed if something about this check is added to the dyno resolver. The thing is, I don't see any reason why this check needs to run in the dyno resolver at all & it makes sense to me for it to be in a later AST transformation pass.